cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -987,6 +1005,12 @@ func (csr *ClusterStateRegistry) GetUpcomingNodes() (upcomingCounts map[string]i
 		ar := csr.acceptableRanges[id]
 		// newNodes is the number of nodes that
 		newNodes := ar.CurrentTarget - (len(readiness.Ready) + len(readiness.Unready) + len(readiness.LongUnregistered))
+		if cloudprovider.IsNodeGroupHibernateEnabled(nodeGroup) {
+			newNodes -= len(readiness.Hibernating)
```

Comment:
How exactly are the hibernated nodes modeled? Here you are subtracting them from `CurrentTarget` here, but in Azure cloudprovider code you write:

> // If the policy for this ScaleSet is Hibernate, the TargetSize is the capacity reported by VMSS minus the nodes
	// in deallocated and deallocating states

Which suggests hibernated nodes are subtracted twice.